src/components/Sidebar.jsx

```diff
@@ -28,6 +37,9 @@ function Sidebar({ user }) {
 
   const { data, isLoading } = useQuery(["userDbList"], getDatabaseList, {
     enabled: !!user,
+    onSuccess: () => {
+      setFirstDatabaseId(data.data.databases[0]._id);
+    },
     onFailure: () => {
       console.log("sending user to errorpage");
     },
```

Comment:
```suggestion
const { data, isLoading } = useQuery(["userDbList"], getDatabaseList, {
    enabled: !!user,
    onSuccess: result => {
      setCurrentDBId(result.data.databases[0]._id);
    },
    onFailure: () => {
      console.log("sending user to errorpage");
    },
    refetchOnWindowFocus: false,
  });
```

유진님 사용자 경험을 생각하여 다른 윈도우 이동시에도 컴포넌트가 계속 focusing 하고 있는 database Id 를 기억할 수 있게 하기 위하여 useEffect와 firstDatabaseId 라는 state를 추가하여 기능 구현 해주신 점 인상 깊네요!! 
다만 실질적으로 sideEffect를 관리하는것이 아닌데 useEffect를 사용하는것은 약간의 로직적인 트릭..? 같은 느낌이 들어 다음과 같이 리팩토링 해보았습니다! 

useQuery의 옵션중 refetchOnWindowFocus 라는 옵션을 꺼주는것으로 윈도우를 이동해도 useQuery는 서버에 fetch 요청을 보내지 않도록 해줄수가 있는데요! 
더하여 firstDatabaseId 라는 state도 사용하지 않을 수 있었습니다! 
이를 통해 고민하던 부분의 해결이 가능했습니다! 
추가적으로 onSuccess시에 data 라는 변수가 undefined 로 나오는 issue가 있어 result 라는 인자로 아이디 값을 업데이트 하도록 변경 했습니다! 
이 부분 한번 고민 해주시면 감사하겠습니다! :)

Reply:
네 최초에 단 한번만 실행되어야 하는 로직을 만들기 위해 머리를 싸맨 결과입니다..! ㅠ dependency로 주는 값을 [0]번째의 id state로 정해줘서, 예를 들어 유저가 맨 처음 db를 삭제하여 [0]번째가 바뀌었을 때만 useEffect내부 setState함수가 돌도록 작성해보았어요. 이 모든게 windowFocus로 인한 refetch문제 때문였구요.
그나저나 refetchOnWindowFocus: false,를 써봤다가 잘 안되서 지우고 다른 방법을 찾으며 시간을 많이 썼는데. onFailure 아래에 위치하면 잘 되나봅니다!!
순서문제였을 수도 있겠다는 깨달음을 얻었네요...

적용해보니 잘 되네요 감사합니다.
리팩토링해서 다시 push하겠습니다!